### ISSUE TITLE
gce-serial: node controller unable to update the node status

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#39639]
The build has been red for a couple of days:
issue: #39406
dashboard: https://k8s-testgrid.appspot.com/google-gce#gce-serial

Many disruptive tests (e.g., cluster resize) are failing because the node controller is unable to update the node status. Here are some relevant log snippets from one such failure:

kube-controller-manager.log:
```
E0104 02:35:53.718936       5 nodecontroller.go:781] Error updating node bootstrap-e2e-minion-group-9qj2: User "system:serviceaccount:kube-system:node-controller" cannot update nodes at the cluster scope.: "<nil>" (put nodes bootstrap-e2e-minion-group-9qj2)
```

kube-apiserver.log:
```
I0104 07:24:48.026670       5 panics.go:75] PUT /api/v1/nodes/bootstrap-e2e-minion-group-9qj2/status: (398.63µs) 403 [[kube-controller-manager/v1.6.0 (linux/amd64) kubernetes/9eaf1aa/system:serviceaccount:kube-system:node-controller] [::1]:44058]
I0104 07:24:48.029180       5 panics.go:75] GET /api/v1/nodes/bootstrap-e2e-minion-group-9qj2: (1.887521ms) 200 [[kube-controller-manager/v1.6.0 (linux/amd64) kubernetes/9eaf1aa/system:serviceaccount:kube-system:node-controller] [::1]:44058]
I0104 07:24:48.050468       5 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:node-controller" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "update" on "nodes./status"
```

Seems to be related to the RBAC work. /cc @deads2k @dchen1107 (build-cop)

**Comment by liggitt** [comment]
opened #39641 to fix

**Comment by yujuhong** [comment]
FYI, below is the "RBAC DENY" error messages from the apiserver.log:
```
  COUNT MESSAGE
      8 rbac.go:49] RBAC DENY: user "admin" groups [system:authenticated] cannot "list" on "pods./"   
      1 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "create" on "nodes./"  
      2 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "create" on "pods./"  
      2 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "get" on "pods./"  
      7 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "list" on "nodes./"  
      6 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "list" on "pods./"  
      6 rbac.go:49] RBAC DENY: user "kubelet" groups [system:nodes system:authenticated] cannot "list" on "services./"  
     12 rbac.go:49] RBAC DENY: user "system:kube-proxy" groups [system:authenticated] cannot "create" on "events./"   
      3 rbac.go:49] RBAC DENY: user "system:kube-proxy" groups [system:authenticated] cannot "list" on "endpoints./"   
      3 rbac.go:49] RBAC DENY: user "system:kube-proxy" groups [system:authenticated] cannot "list" on "services./"   
   2323 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:node-controller" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "delete" on "nodes./" 
   2480 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:node-controller" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "list" on "pods./" 
  12400 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:node-controller" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "update" on "nodes./status" 
   1079 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:persistent-volume-binder" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "watch" on "events./" 
    889 rbac.go:49] RBAC DENY: user "system:serviceaccount:kube-system:pod-garbage-collector" groups [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] cannot "list" on "nodes./" 
```

Some of them may be transient (?) or irrelevant, but the messages with high counts are probably worth looking.

**Comment by liggitt** [comment]
updated the PR with the permissions used in the nodecontroller code. will look at the pv and garbage collection denials separately

**Comment by yujuhong** [comment]
Should be fixed by @liggitt's PR #39641. Closing the issue.
